src/api/style/style.api.ts

```diff
@@ -97,26 +97,40 @@ export async function submitStyleRewrite(
 export async function styleCheck(
   styleAnalysisRequest: StyleAnalysisReq,
   config: Config,
+  options: AnalysisWorkflowOptions = defaultAnalysisWorkflowOptions,
```

Comment:
I think we  should extend the `config` and add timeout to it instead of creating a new options param in `styleCheck` The config already has stuff like `rateLimit` so adding timeout to it should also be fine

Reply:
Sorry for taking so long. I extended `config` by creating another type of config named `WorkflowConfig` and used it for the `styleCheck`, `styleSuggestion` and `styleRewrite`. It did not add it to the regular config, because then the timeout option would be everywhere, also where we don't need it. If I would have extended the regular config then we would have timeout also twice for batchCheck:
```
export function styleBatchCheckRequests(
  requests: StyleAnalysisReq[],
  config: Config,
  options: BatchOptions = {},
  ...
 ```
 once in config and once in BatchOptions.